login.php

```diff
@@ -62,9 +66,14 @@
         $name = $_POST['name'];
         $password = $_POST['password'];
 
+		$hashedpw= $db->getRow("SELECT `password` FROM `".MLS_PREFIX."users` WHERE `username` = ?s", $_POST['name']);
 
-        if(!($usr = $db->getRow("SELECT `userid` FROM `".MLS_PREFIX."users` WHERE `username` = ?s AND `password` = ?s", $name, sha1($password))))
-            $page->error = "Username or password are wrong !";
+        if(!($usr = $db->getRow("SELECT `userid` FROM `".MLS_PREFIX."users` WHERE `username` = ?s", $_POST['name'])) || !(password_verify($password, ($hashedpw->password)))){            
+	     $page->error = "Username or password are wrong !";
+		}
+		else if($_POST)
+			if(!$captcha)
+				$page->error = "Invalid captcha code !";
         else {
             if($_POST['r'] == 1){
                 $path_info = parse_url($set->url);
```

Comment:
In this statement `!(password_verify($password, ($hashedpw->password)))`, we are comparing the password provided by the user with the hashed password stored in the database. password_verify takes plaintext password and compares with the hashed password (in your database) and returns boolean - true or false, so we don't have to hash the password ourselves.